packages/react/src/Hero/Hero.tsx

```diff
@@ -1,69 +1,94 @@
-import React, {forwardRef} from 'react'
+import React, {forwardRef, PropsWithChildren, useMemo} from 'react'
 import clsx from 'clsx'
 import styles from './Hero.module.css'
-import {Button} from '../Button'
-import {Heading} from '../Heading'
+import {Button, ButtonBaseProps} from '../Button'
+import {Heading, HeadingProps} from '../Heading'
 import {Text} from '../Text'
 
 import type {BaseProps} from '../component-helpers'
 import '@primer/brand-primitives/lib/design-tokens/css/tokens/functional/components/hero/base.css'
 
-type Action = {
-  text: string
+export type HeroProps = BaseProps<HTMLDivElement> & {
+  align?: 'start' | 'center'
+  children: React.ReactNode | React.ReactNode[] // Maybe use this from `FAQ`? type FAQRootProps = PropsWithChildren<BaseProps<HTMLElement>>
+}
+
+const Root = forwardRef<HTMLDivElement, HeroProps>(({className, align = 'start', children, ...rest}, ref) => {
+  const HeroActions = useMemo(
+    () =>
+      React.Children.toArray(children).filter(
+        child => React.isValidElement(child) && (child.type === HeroPrimaryAction || child.type === HeroSecondaryAction)
+      ),
+    [children]
+  )
+
+  return (
+    <section
+      className={clsx(styles.Hero, styles[`Hero--align-${align}`], className)}
+      ref={ref}
+      aria-labelledby="hero-section-brand-heading"
+      {...rest}
+    >
+      {React.Children.map(children, child => {
```

Comment:
could we hoist these outside the render and memoize them like you did for buttons? And when that happens, wonder if you could do both filters in one loop rather than 2. That would be a neat optimisation, as they are otherwise doing the same thing but with inverted conditions.

Reply:
Yup! Replaced this with a `.reduce` so that we'd only run one loop. Let me know if this isn't the right way to go about it!